### ISSUE TITLE
Building static library for Conan produces a version stub instead of the corresponding DuckDB version

### ISSUE THREAD

**Issue by abba-kovner** [duckdb/duckdb#10847]
### What happens?

Conan is a software package manager for C++. Specifically, for DuckDB it uses the dedicate [recipe](https://github.com/conan-io/conan-center-index/blob/master/recipes/duckdb/all/conanfile.py) (a Python file) to configure and build the DuckDB project. Currently, the generated static library for Linux contains the stub `0.0.1-dev0` version. For the latest DuckDB version that is present on Conan, i.e. [0.9.2](https://conan.io/center/recipes/duckdb?version=0.9.2), the DuckDB inside the generated static library should be `0.9.2` as well. Otherwise the incorrect versioning impedes using DuckDB extensions, e.g. the SQLite extension.

The screenshot illustrates the generated DuckDB version in the static library file `libduckdb_static.a`:
![libduckdb_static](https://github.com/duckdb/duckdb/assets/132137032/fa41864f-cc8d-4954-9b22-b684a0b54176)
Please see the bottom right corner.

### To Reproduce

1. `pip install conan`
2. `conan profile new default --detect`
3. `echo [requires] >> conanfile.txt`
4. `echo duckdb/0.9.2 >> conanfile.txt`

Add the following to `CMakeLists.txt`:
```
if (NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
    message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
    file(DOWNLOAD "https://raw.githubusercontent.com/conan-io/cmake-conan/0.18.1/conan.cmake" "${CMAKE_BINARY_DIR}/conan.cmake" TLS_VERIFY ON)
endif ()

include(${CMAKE_BINARY_DIR}/conan.cmake)
                                                                                                                                                                              
conan_cmake_install(PATH_OR_REFERENCE ../conanfile.txt BUILD missing PROFILE default)
```

### OS:

x86_64 GNU/Linux

### DuckDB Version:

0.9.2

### DuckDB Client:

C++

### Full Name:

Igor Bronshtein

### Affiliation:

MSAID GmbH

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a release build (and could not test with a nightly build)

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Mause** [comment]
Hi there, we don't actually maintain the Conan package. Is this something you're expecting us to fix on the DuckDB side?

**Comment by carlopi** [comment]
I think this needs to be raised with the conan maintainers, and there have been recently some changes that should make it work correctly for version 0.10.1/0.10.2 (I think).

Can you try whether more recent versions work correctly ?

**Comment by abba-kovner** [comment]
Fixed for `duckdb/1.0.0` in the Conan repository
https://conan.io/center/recipes/duckdb
![image](https://github.com/user-attachments/assets/753a4f84-3b25-49aa-8dde-9666cd992ac6)
